WordPress/src/main/java/org/wordpress/android/ui/posts/photochooser/PhotoChooserFragment.java

```diff
@@ -0,0 +1,293 @@
+package org.wordpress.android.ui.posts.photochooser;
+
+import android.app.Fragment;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.v4.app.ActivityCompat;
+import android.support.v4.app.ActivityOptionsCompat;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.view.ActionMode;
+import android.support.v7.widget.GridLayoutManager;
+import android.support.v7.widget.RecyclerView;
+import android.view.LayoutInflater;
+import android.view.Menu;
+import android.view.MenuInflater;
+import android.view.MenuItem;
+import android.view.View;
+import android.view.ViewGroup;
+
+import org.wordpress.android.R;
+import org.wordpress.android.ui.posts.EditPostActivity;
+import org.wordpress.android.ui.posts.photochooser.PhotoChooserAdapter.OnAdapterLoadedListener;
+import org.wordpress.android.util.AniUtils;
+
+import java.util.ArrayList;
+
+public class PhotoChooserFragment extends Fragment {
+
+    static final int NUM_COLUMNS = 3;
+
+    enum PhotoChooserIcon {
+        ANDROID_CAMERA,
+        ANDROID_PICKER,
+        WP_MEDIA
+    }
+
+    public interface OnPhotoChooserListener {
+        void onItemTapped(Uri mediaUri);
+        void onItemDoubleTapped(View view, Uri mediaUri);
+        void onSelectedCountChanged(int count);
+    }
+
+    private RecyclerView mRecycler;
+    private PhotoChooserAdapter mAdapter;
+    private View mBottomBar;
+    private ActionMode mActionMode;
+    private GridLayoutManager mGridManager;
+    private Parcelable mRestoreState;
+
+    public static PhotoChooserFragment newInstance() {
+        Bundle args = new Bundle();
+        PhotoChooserFragment fragment = new PhotoChooserFragment();
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        // retain the instance state since EditPostActivity has android:configChanges in the manifest
+        setRetainInstance(true);
```

Comment:
You're right - `setRetainInstance` really isn't necessary. I was overthinking the impact of the activity relying on `configChanges`. I've removed `setRetainInstance` in e2ba331.

Note that I talked with the Aztec folks and it sounds like at some point `configChanges` will be dropped since the Aztec editor doesn't require it. If and when that happens, we'll revisit how to retain this fragment's state.